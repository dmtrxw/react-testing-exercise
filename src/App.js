import React from 'react';
import { Switch, Route, Link } from 'react-router-dom';

import Home from './Home';
import TodoList from './TodoList';

function App() {
  return (
    <>
      <h1>Todo App</h1>
      <nav>
        <Link to="/" data-testid="link-to-home">Home</Link>
        <Link to="/todos" data-testid="link-to-todos">Todos</Link>
      </nav>
      <hr />
      <Switch>
        <Route exact path="/">
          <Home />
        </Route>
        <Route path="/todos">
          <TodoList />
        </Route>
      </Switch>
    </>
  );
}

export default App;
