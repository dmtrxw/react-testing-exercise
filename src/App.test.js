import React from 'react';
import { render } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';

import App from './App';

test('App should render', () => {
  const app = render(
    <Router>
      <App />
    </Router>
  );

  expect(app.queryByText(/Welcome/)).toBeInTheDocument();
  expect(app.queryByText(/Your todos/)).not.toBeInTheDocument();
  expect(app.queryByTestId('link-to-home')).toBeInTheDocument();
  expect(app.queryByTestId('link-to-todos')).toBeInTheDocument();
});

test('App should render and matches the snapshot', () => {
  const app = render(
    <Router>
      <App />
    </Router>
  );
  expect(app).toMatchSnapshot();
});
