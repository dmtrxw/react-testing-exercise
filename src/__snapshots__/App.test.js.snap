// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render and matches the snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <h1>
        Todo App
      </h1>
      <nav>
        <a
          data-testid="link-to-home"
          href="/"
        >
          Home
        </a>
        <a
          data-testid="link-to-todos"
          href="/todos"
        >
          Todos
        </a>
      </nav>
      <hr />
      <p>
        Welcome to Todo App, manage your tasks easily!
      </p>
    </div>
  </body>,
  "container": <div>
    <h1>
      Todo App
    </h1>
    <nav>
      <a
        data-testid="link-to-home"
        href="/"
      >
        Home
      </a>
      <a
        data-testid="link-to-todos"
        href="/todos"
      >
        Todos
      </a>
    </nav>
    <hr />
    <p>
      Welcome to Todo App, manage your tasks easily!
    </p>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
